File: app/src/main/java/br/com/disapps/homepet/ui/hotel/adapter/HotelViewPagerAdapter.java
package br.com.disapps.homepet.ui.hotel.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;


import br.com.disapps.homepet.ui.comments.CommentsFragment;
import br.com.disapps.homepet.ui.details.HotelDetailsFragment;
import br.com.disapps.homepet.ui.map.MapFragment;

/**
 * Created by diefferson.santos on 31/08/17.
 */

public class HotelViewPagerAdapter  extends FragmentPagerAdapter {

    private static final int FRAGMENT_COUNT = 3;
    private final Context mContext;
    private final int codeHotel;

    public HotelViewPagerAdapter(FragmentManager fm, Context c, int codeHotel) {
        super(fm);
        mContext = c;
        this.codeHotel = codeHotel;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position){
            case 0:
                return HotelDetailsFragment.newInstance(codeHotel);
            case 1:
                return CommentsFragment.newInstance(codeHotel);
            case 2:
                return MapFragment.newInstance(codeHotel);
        }

       // return LectureDetailsFragment.newInstance();
        return null;
    }

    @Override
    public int getCount() {
        return FRAGMENT_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {

        switch (position){
            case 0:
                return "Detalhes";
            case 1:
                return "Comentarios";
            case 2:
                return "Mapa";
        }

        return null;
    }
}

File: app/src/main/java/br/com/disapps/homepet/ui/hotel/adapter/HotelViewPagerAdapter.kt
package br.com.disapps.homepet.ui.hotel.adapter

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter


import br.com.disapps.homepet.ui.comments.CommentsFragment
import br.com.disapps.homepet.ui.details.HotelDetailsFragment
import br.com.disapps.homepet.ui.map.MapFragment

/**
 * Created by diefferson.santos on 31/08/17.
 **/

class HotelViewPagerAdapter(fm: FragmentManager, private val mContext: Context, private val codeHotel: Int) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment? {
        when (position) {
            0 -> return HotelDetailsFragment.newInstance(codeHotel)
            1 -> return CommentsFragment.newInstance(codeHotel)
            2 -> return MapFragment.newInstance(codeHotel)
        }

        // return LectureDetailsFragment.newInstance();
        return null
    }

    override fun getCount(): Int {
        return FRAGMENT_COUNT
    }

    override fun getPageTitle(position: Int): CharSequence? {

        when (position) {
            0 -> return "Detalhes"
            1 -> return "Comentarios"
            2 -> return "Mapa"
        }

        return null
    }

    companion object {

        private val FRAGMENT_COUNT = 3
    }
}
